### PR TITLE
Allow to skip format with simple command.

### PR DESCRIPTION
**Pull request by The-Huginn**
Please review @xstefank

### CODE REVIEW

File: src/main/java/io/xstefank/wildfly/bot/PullRequestFormatProcessor.java [hunk 1]
```diff
@@ -60,6 +61,12 @@ void verifyFormat(@PullRequest.Edited @PullRequest.Opened @PullRequest.Synchroni
         GHPullRequest pullRequest = pullRequestPayload.getPullRequest();
         Map<String, String> errors = new HashMap<>();
 
+        if (Pattern.compile("@%s skip format".formatted(wildFlyBotConfig.githubName()), Pattern.DOTALL | Pattern.CASE_INSENSITIVE | Pattern.LITERAL)
```
Comment (xstefank) [inline src/main/java/io/xstefank/wildfly/bot/PullRequestFormatProcessor.java:64]: move pattern compile to field, we don't need to recompile the pattern on every invocation

Comment (xstefank) [reply src/main/java/io/xstefank/wildfly/bot/PullRequestFormatProcessor.java:64]: isn't the case insensitive possible issue with `@bot-nAmE` mentions? I believe that `@mention` is not case insensitive.

File: src/main/java/io/xstefank/wildfly/bot/PullRequestFormatProcessor.java [hunk 2]
```diff
@@ -60,6 +61,12 @@ void verifyFormat(@PullRequest.Edited @PullRequest.Opened @PullRequest.Synchroni
...
+                .matcher(pullRequest.getBody()).find()) {
+            LOG.info("Skipping format");
```
Comment (xstefank) [inline src/main/java/io/xstefank/wildfly/bot/PullRequestFormatProcessor.java:66]: for PR...


keep in mind that we read this in the bot log so how should we associate this log message with anything?

**Review by xstefank — CHANGES_REQUESTED** [review]

File: src/main/java/io/xstefank/wildfly/bot/PullRequestFormatProcessor.java (hunk 1 above)
Comment (The-Huginn) [reply src/main/java/io/xstefank/wildfly/bot/PullRequestFormatProcessor.java]: true but I wanted to do is just as temporary thing anyways. I can remove the insensitivity if you want

**Comment by The-Huginn** [comment]
Additionally I have added changes to resolve  #145 as well

File: src/main/java/io/xstefank/wildfly/bot/PullRequestFormatProcessor.java [hunk 3]
```diff
@@ -161,4 +163,27 @@ private List<Check> initializeChecks(WildFlyConfigFile wildflyConfigFile) {
 
         return checks;
     }
+
+    @PostConstruct
+    void construct() {
+        SKIP_FORMAT_COMMAND = Pattern.compile("@%s skip format".formatted(wildFlyBotConfig.githubName()), Pattern.DOTALL | Pattern.CASE_INSENSITIVE | Pattern.LITERAL);
```
Comment (xstefank) [inline src/main/java/io/xstefank/wildfly/bot/PullRequestFormatProcessor.java:169]: you can do this in the declaration of the field, no need for post construct method, and drop case insesitivity

File: src/main/java/io/xstefank/wildfly/bot/PullRequestFormatProcessor.java [hunk 4]
```diff
@@ -161,4 +163,27 @@ private List<Check> initializeChecks(WildFlyConfigFile wildflyConfigFile) {
...
+    }
+
+    private boolean skipPullRequestFormat(GHPullRequest pullRequest, WildFlyConfigFile wildFlyConfigFile) throws IOException {
+        if (SKIP_FORMAT_COMMAND.matcher(pullRequest.getBody()).find()) {
+            LOG.infof("Skipping format for following Pull Request [#%d] - %s", pullRequest.getNumber(), pullRequest.getTitle());
+            return true;
+        }
+
+        if (wildFlyConfigFile == null) {
+            LOG.error("No configuration file available. ");
```
Comment (xstefank) [inline src/main/java/io/xstefank/wildfly/bot/PullRequestFormatProcessor.java:179]:
```suggestion
            LOG.error("No configuration file available.");
```

File: src/main/java/io/xstefank/wildfly/bot/PullRequestFormatProcessor.java [hunk 5]
```diff
@@ -161,4 +163,27 @@ private List<Check> initializeChecks(WildFlyConfigFile wildflyConfigFile) {
...
+            return true;
+        }
+
+        if (pullRequest.isDraft()) {
```
Comment (xstefank) [inline src/main/java/io/xstefank/wildfly/bot/PullRequestFormatProcessor.java:183]: also log?

File: src/main/java/io/xstefank/wildfly/bot/PullRequestFormatProcessor.java (hunk 4 above)
Comment (xstefank) [reply src/main/java/io/xstefank/wildfly/bot/PullRequestFormatProcessor.java:179]: also add  PR reference

File: src/main/java/io/xstefank/wildfly/bot/TriagePullRequestProcessor.java [hunk 6]
```diff
@@ -57,5 +56,18 @@ void onPullRequestOpened(@PullRequest.Opened GHEventPayload.PullRequest pullRequ
...
+            LOG.error("No configuration file available.");
+            return true;
+        }
+
+        if (pullRequest.isDraft()) {
+            return true;
+        }
+
+        return false;
+    }
```
Comment (xstefank) [inline src/main/java/io/xstefank/wildfly/bot/TriagePullRequestProcessor.java:71]: same comments for log messages

File: src/test/java/io/xstefank/wildfly/bot/PRSkipPullRequestTest.java [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+                });
+    }
+
+    @Test
+    void testSkippingFormatCheckOnDraft() throws IOException {
+        gitHubJson = GitHubJson.builder(VALID_PR_TEMPLATE_JSON)
+                .title(INVALID_TITLE)
+                .build();
+        given().github(mocks -> {
+            // TODO update with `add-labels` branch merged
```
Comment (xstefank) [inline src/test/java/io/xstefank/wildfly/bot/PRSkipPullRequestTest.java:69]: do we need this? You will rebase it yourself

**Review by xstefank — CHANGES_REQUESTED** [review]

File: src/test/java/io/xstefank/wildfly/bot/PRSkipPullRequestTest.java (hunk 7 above)
Comment (The-Huginn) [reply src/test/java/io/xstefank/wildfly/bot/PRSkipPullRequestTest.java:69]: Yes, that's a reminder for myself for that rebasing

**Review by xstefank — APPROVED** [review]

**Pull request merged**